univention/udm_share.py

```diff
@@ -0,0 +1,530 @@
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.univention_umc import (
+    umc_module_for_add,
+    umc_module_for_edit,
+    ldap_search,
+    base_dn,
+)
+import socket
+
+
+DOCUMENTATION = '''
+---
+module: udm_share
+version_added: "2.2"
+author: "Tobias Rueetschi (@2-B)"
+short_description: Manage samba shares on a univention corporate server
+description:
+    - "This module allows to manage samba shares on a univention corporate
+       server (UCS).
+       It uses the python API of the UCS to create a new object or edit it."
+requirements:
+    - Python >= 2.6
+options:
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the share is present or not.
+    name:
+        required: true
+        description:
+            - Name
+    host:
+        required: false
+        default: None
+        description:
+            - Host FQDN (server which provides the share), e.g. C({{
+              ansible_fqdn }}). Required if C(state=present).
+    path:
+        required: false
+        default: None
+        description:
+            - Directory on the providing server, e.g. C(/home). Required if C(state=present).
+    sambaName:
+        required: false
+        default: None
+        description:
+            - Windows name. Required if C(state=present).
+    ou:
+        required: true
+        description:
+            - Organisational unit, inside the LDAP Base DN.
+    owner:
+        required: false
+        default: 0
+        description:
+            - Directory owner of the share's root directory.
+    group:
+        required: false
+        default: '0'
+        description:
+            - Directory owner group of the share's root directory.
+    directorymode:
+        required: false
+        default: '00755'
+        description:
+            - Permissions for the share's root directory.
+    root_squash:
+        required: false
+        default: '1'
+        choices: [ '0', '1' ]
+        description:
+            - Modify user ID for root user (root squashing).
+    subtree_checking:
+        required: false
+        default: '1'
+        choices: [ '0', '1' ]
+        description:
+            - Subtree checking.
+    sync:
+        required: false
+        default: 'sync'
+        description:
+            - NFS synchronisation.
+    writeable:
+        required: false
+        default: '1'
+        choices: [ '0', '1' ]
+        description:
+            - NFS write access.
+    sambaBlockSize:
```

Comment:
Yes, I understand that. This is because the python API of univention is like that. The parameters of the ansible module uses the same name as the API provides.

Line 485 - 490:
```Python
+            for k in obj.keys():
+                if module.params[k] is True:
+                    module.params[k] = '1'
+                elif module.params[k] is False:
+                    module.params[k] = '0'
+                obj[k] = module.params[k]
```

So, that's why I mixed camelCase and snake_case.